File: src/lib/types/tag.ts
/**
 * タグ
 */
export interface Tag {
  /** タグID */
  id: string;
  /** タグ名 */
  name: string;
  /** タグの色 */
  color?: string;
  /** 表示順序 */
  orderIndex?: number;
  /** 作成日時 */
  createdAt: Date;
  /** 更新日時 */
  updatedAt: Date;
  /** 削除フラグ（論理削除） */
  deleted: boolean;
  /** 最終更新者のユーザーID */
  updatedBy: string;
}

/**
 * タグ検索条件
 */
export interface TagSearchCondition {
  /** タグ名 */
  name?: string;
}

File: src-tauri/src/models/command/tag.rs
use serde::{Serialize, Deserialize};

use crate::models::{command::ModelConverter, tag::Tag};

/// Tauriコマンド引数用のTag構造体（created_at/updated_atはString）
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TagCommand {
    pub id: String,
    pub name: String,
    pub color: Option<String>,
    pub order_index: Option<i32>,
    pub created_at: String,
    pub updated_at: String,
}

impl ModelConverter<Tag> for TagCommand {
    /// コマンド引数用（TagCommand）から内部モデル（Tag）に変換
    async fn to_model(&self) -> Result<Tag, String> {
        use chrono::{DateTime, Utc};

        let created_at = self.created_at.parse::<DateTime<Utc>>()
            .map_err(|e| format!("Invalid created_at format: {}", e))?;
        let updated_at = self.updated_at.parse::<DateTime<Utc>>()
            .map_err(|e| format!("Invalid updated_at format: {}", e))?;

        use crate::types::id_types::TagId;
        
        Ok(crate::models::tag::Tag {
            id: TagId::from(self.id.clone()),
            name: self.name.clone(),
            color: self.color.clone(),
            order_index: self.order_index,
            created_at,
            updated_at,
        })
    }
}

/// Tauriコマンド引数用のTagSearchRequest構造体
#[derive(Debug, Serialize, Deserialize)]
pub struct TagSearchRequest {
    pub name: Option<String>,
    pub color: Option<String>,
    pub created_from: Option<String>,
    pub created_to: Option<String>,
    pub usage_count_min: Option<u32>,
    pub limit: Option<usize>,
    pub offset: Option<usize>,
    pub order_by_popularity: Option<bool>,
}
